codemodel/model/common/schemas.ts

```diff
@@ -126,6 +128,23 @@ export class Schemas {
     }
 
     const a: Array<Schema> = ((<any>this)[group] || ((<any>this)[group] = new Array<Schema>()));
+
+    // for simple types, go a quick check to see if an exact copy of this is in the collection already 
+    // since we can just return that. (the consumer needs to pay attention tho')
```

Comment:
Any way we can add this to the verbose log so that it's more obvious when it happens?  Otherwise I don't think the consumer would have any way of knowing it got duped out.

Reply:
The 'consumer' is modelerfour (this should be transparent to the user). 

I'm actually writing a checker plugin right this second to do a sanity check on the code model itself that can yell when something isn't quite right.